format_spec.md

```diff
@@ -313,8 +313,23 @@ This field MUST be either an Array or String. If this field is an array, it MUST
 
 ##### `src` (optional)
 
-If not all information is listed inside a dependency object and it's metadata is not hosted on the same host which is serving metadata for the parent project, the `src` field may be used. This field MUST contain a URL. This URL MUST serve MODIP Format-compliant metadata about the required dependency. It MUST include the `versions` field.  
-    
+This specifies where information about this dependency can be acquired, but it's not always required. If the metadata of the parent project is being served from a host, then this field is not required and clients can lookup the ID at the host. For example, if I request metadata about a project from Modrinth and there is a dependency, "my-dependency", then I should call Modrinth's API in order to find out more about that dependency.
```

Comment:
Does this mean you can't point a client to a MODIP file URL and expect it to just work? The client must have specific support for the server

Reply:
The reasoning is: if your launcher has integration for a host, you don’t need to include a URL as the launcher already knows where to get metadata from. But if you’re doing something *not integrated*, like clicking a “download MODIP metadata” button on a host, the `src` field needs to be there as there is no direct tie between the metadata and the host, unlike with direct integration.

But I see the confusion, and I can see this being misused. I’ll consider requiring it or reforming it somehow.

Reply:
Wouldn't you expect to get the same file no matter how you download it?

Reply:
Ideally, you would get the same information, but just split up into “chunks” using the host’s API. This also delves into the problem of APIs and splitting up calls so you aren’t sending a humongous JSON file when you only need a bit of it, which is a separate issue but still important.

Reply:
I see. Mark this resolved.